### ISSUE TITLE
Write longer struct names, use modules for namespacing instead of writing `Tpm2` or `Tpms`

### ISSUE THREAD

**Issue by kupiakos** [tpm-rs/tpm-rs#15]
`Tpm2b*` has a specific meaning, but many of these can be generic types or modules instead of separate, opaque, and self-namespacing struct names.

There's a good argument for keeping with the spec, but I'd like for us to balance that with using more of Rust's advanced features for better readability.

**Comment by trent-reed** [comment]
On one hand, it does seem natural to want to namespace things like `tpm2::tpms::AlgProperty`. On the other hand I'd hate to force folks to do `use tpm2::tpms::AlgProperty as TpmsAlgProperty` when they inevetably conflict with `tpm2::tpml::AlgProperty` (or other). This will basically lead to most TPM developers as doing `use tpm2::tpml` and then using `tpml::AlgProperty`, but if that's the case why even have the separate module?

Also, I'm not necessarily in love with the idea of allowing folks doing `use tpm2::tpms::AlgProperty` when there's basically two different `AlgProperty` types. This feels very easy for folks to want to do, and very much lacks readability.

The fact that the types are named `<prefix><type>` is an organization strategy that makes sense for the spec, but in natural language what these types actually want to say is something like `tpms::AlgProperty as AlgProperty` (structure) and `tpml::AlgProperty as AlgProperties` (list), but that kind of naming is tough to build standards around - so I definitely don't fault the spec naming choice here.

I personally feel pretty strongly against this issue:

1. I don't want to encourage `use tpm2::tpms::AlgProperty`.
2. I value alignment with the specification over client-visible module nesting. The TPM is already a complicated piece of technology, aligning as close as possible with the spec on naming for type names is probably a good thing (because I suspect almost everyone will be refering to the spec while doing TPM programming).
3. The prefix says something about the type, something interesting that we lose when we chop it off the base typename (it's a "list", it's a "structure", it's a "value", etc). This is something that normally would be expressed in different naming conventions, but we work with what we have.

Note that we can still organize into submodules to aide in developing our own crate so we aren't just staring at one huge `lib.rs`, I've made such suggestions here: 

*   https://github.com/orgs/tpm-rs/discussions/18
*   https://github.com/orgs/tpm-rs/discussions/19

**Comment by bradlitterell** [comment]
@trent-reed @kupiakos  @vsukhoml - can we close this issue in light of #97?

**Comment by trent-reed** [comment]
I think we can probably dupe to that.

TL;DR that covers the module topology, and we've mostly been of the opinion that we don't want to strip the prefixes from the typenames, but that we're fine with separating into submodules and re-exporting in order to keep the project maintainable.

DUPE: #97
